django/applications/catmaid/control/annotation.py

```diff
@@ -1156,8 +1156,9 @@ def generate_annotation_intersection_query(project_id, annotations):
 
     return q
 
-
-def generate_co_annotation_query(project_id, co_annotation_ids, classIDs, relationIDs):
+def generate_co_annotation_query(project_id:Union[int,str], co_annotation_ids, classIDs, relationIDs):
+    # XXX Unclear if the below early return will create problems for the caller. If we could remove that (or convert it to an
+    # exception) the function would have the type of Tuple[str,str]
```

Comment:
Yes, you are right. Returning early will indeed create problems (`TypeError: 'NoneType' object is not iterable`). In the current call hierarchy it is made sure that `co_annotation_ids` exist, but it is still not a nice interface. I believe an exception like `raise ValueError("Need co-annotations")` or similar is the right way to go here. Like I said, it's checked before, so it shouldn't have an impact.